src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -1186,7 +1203,7 @@ fn handle_message<K, V, T, D>(
     };
 
     let ts_cap = cap_set.delayed(&ts);
-    output_handle.session(&ts_cap).give(output);
+    output_handle.session(&ts_cap).give((output, ts, diff));
```

Comment:
its nice here that we are moving the diff out of the `output`, but also kindof unfortunate we have to copy the `ts` into the value. Its always felt a bit weird that `Collection` forces you to have the `ts` in the data as well, but its not too onerous for us i guess...